src/third_party/progress/progress.hpp

```diff
@@ -57,6 +57,10 @@ class C_Progress
       if ( !_expected_count ) _expected_count = 1;  // prevent divide by zero
     } // restart
 
+    /** @brief Indicator if the current operation should be aborted.
+     * @return Return true if the process has been canceled by the user.
+     **/
+    virtual bool hasBeenCanceled()const { return false; }
```

Comment:
Since we introduce a new comportment it would be nice to add a unit test to check that we can stop a loop